manager/role_manager.go

```diff
@@ -100,6 +139,51 @@ func (rm *roleManager) Run(ctx context.Context) {
 	}
 }
 
+// evictRemovedNode evicts a removed node from the raft cluster membership.  This is to cover an edge case in which
+// a node might have been removed, but somehow the role was not reconciled (possibly a demotion and a removal happened
+// in rapid succession before the raft membership configuration went through).
+func (rm *roleManager) evictRemovedNode(ctx context.Context, raftMember *api.RaftMember) {
+	// Check if the member still exists in the membership
+	member := rm.raft.GetMemberByNodeID(raftMember.NodeID)
+	if member != nil {
+		// We first try to remove the raft node from the raft cluster.  On the next tick, if the node
```

Comment:
Why can't we do it in the same tick ?

Reply:
Honestly I'm not sure, as I am not super familiar with this code either - this was just how the logic was previously for demotion.  I think we probably can just update the node immediately, since if we remove the node, eventually https://github.com/docker/swarmkit/blob/master/manager/state/raft/raft.go#L1850 gets called, which seems to wait for the configuration change to be processed and applied to the server before returning.  Does that sound right cc @aaronlehmann?

Reply:
(As the code is written, this allows a node reconciliation to be retried if there's a logged error, but the code can be refactored.  Not sure if we require that updating the node objects happen in the next tick on successful cluster membership removal, though)

Reply:
I'm not completely sure but it may just be to simplify error handling. If we did it in the same tick, we'd have to make sure it was actually removed before deleting it from the list. I don't think there's any downside to deferring the removal from the list.

Reply:
@aaronlehmann Cool, thanks!  I agree I think there's no downside.  I just wanted to be able to add a comment as to why it is deferred until the next tick.

Reply:
This is probably bikeshedding, but this makes the code somewhat harder to read and reason with. I agree that there might not be a downside to doing it this way, but I'm not sure that's motivation enough. Thats my take, but I'll let you decide :) @cyli

Reply:
@anshulpundir If we were 100% sure that error handling is the only reason, I'd be happy to make the change.  As it is I think I'm only pretty sure it it can be updated in the same tick, so perhaps not making the change would be safer.  I think I'd want more comprehensive tests to make sure that making that change wouldn't break a corner case I didn't understand before I make that change.  

I will see if I can figure out how to test that if removing a node errors, that the removal will be retried later, though.  I haven't added that cases because I couldn't think of a good way to write that test yet.  (I don't think even if this were added it'd be sufficient to guarantee I didn't break something with the refactor).

Reply:
> If we were 100% sure that error handling is the only reason, I'd be happy to make the change. 

I guess we're not, so I'm OK with leaving this as is for now.